pkg/controllers/walruscore/connector.go

```diff
@@ -3,29 +3,142 @@ package walruscore
 import (
 	"context"
 
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
 	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlcli "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlevent "sigs.k8s.io/controller-runtime/pkg/event"
+	ctrlhandler "sigs.k8s.io/controller-runtime/pkg/handler"
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 	ctrlreconcile "sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	walruscore "github.com/seal-io/walrus/pkg/apis/walruscore/v1"
+	"github.com/seal-io/walrus/pkg/apistatus"
 	"github.com/seal-io/walrus/pkg/controller"
+	"github.com/seal-io/walrus/pkg/resourcehandler"
+	"github.com/seal-io/walrus/pkg/resourcehandlers"
+	"github.com/seal-io/walrus/pkg/systemmeta"
 )
 
 // ConnectorReconciler reconciles a v1.Connector object.
-type ConnectorReconciler struct{}
+type ConnectorReconciler struct {
+	client ctrlcli.Client
+}
 
 var _ ctrlreconcile.Reconciler = (*ConnectorReconciler)(nil)
 
 func (r *ConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = ctrllog.FromContext(ctx)
+	logger := ctrllog.FromContext(ctx)
+
+	obj := new(walruscore.Connector)
+	err := r.client.Get(ctx, req.NamespacedName, obj)
+	if err != nil {
+		return ctrl.Result{}, ctrlcli.IgnoreNotFound(err)
+	}
+
+	// Skip.
+	{
+		// Skip deletion.
+		if obj.DeletionTimestamp != nil {
+			return ctrl.Result{}, nil
+		}
+
+		// Skip connector already reconciling.
+		if apistatus.ConnectorConditionConnected.IsUnknown(obj) && !apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			return ctrl.Result{}, nil
+		}
+
+		// Skip connector finished reconciling.
+		if (apistatus.ConnectorConditionConnected.IsTrue(obj) || apistatus.ConnectorConditionConnected.IsFalse(obj)) &&
+			!apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// Initialize status.
+	{
+		apistatus.ConnectorConditionConnected.Unknown(obj, apistatus.ConnectorConditionReasonConnecting, "Connecting")
+		if apistatus.ConnectorConditionRefresh.IsUnknown(obj) {
+			apistatus.ConnectorConditionRefresh.False(obj, "", "")
+		}
+		err = r.client.Status().Update(ctx, obj)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
use predication to prevent a new enqueue event after calling `connectors/status` api.